bpdm-cleaning-service-dummy/src/main/kotlin/org/eclipse/tractusx/bpdm/cleaning/service/CleaningServiceDummy.kt

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.cleaning.service
+
+
+import mu.KotlinLogging
+import org.eclipse.tractusx.bpdm.common.dto.AddressType
+import org.eclipse.tractusx.orchestrator.api.client.OrchestrationApiClient
+import org.eclipse.tractusx.orchestrator.api.model.*
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+import java.util.*
+
+@Service
+class CleaningServiceDummy(
+    private val orchestrationApiClient: OrchestrationApiClient,
+
+    ) {
+
+    private val logger = KotlinLogging.logger { }
+
+
+    @Scheduled(cron = "\${cleaningService.pollingCron:-}", zone = "UTC")
+    fun pollForCleaningTasks() {
+        try {
+            logger.info { "Starting polling for cleaning tasks from Orchestrator..." }
+
+            // Step 1: Fetch and reserve the next cleaning request
+            val cleaningRequest = orchestrationApiClient.goldenRecordTasks
+                .reserveTasksForStep(TaskStepReservationRequest(amount = 10, TaskStep.Clean))
+
+            val cleaningTasks = cleaningRequest.reservedTasks
+
+            logger.info { "${cleaningTasks.size} tasks found for cleaning. Proceeding with cleaning..." }
+
+            if (cleaningTasks.isNotEmpty()) {
+
+                val cleaningResults = cleaningTasks.mapNotNull { reservedTask ->
+                    // Step 2: Generate dummy cleaning results
+                    processCleaningTask(reservedTask)
+                }
+
+                // Step 3: Send the cleaning result back to the Orchestrator
+                orchestrationApiClient.goldenRecordTasks.resolveStepResults(TaskStepResultRequest(cleaningResults))
+                logger.info { "Cleaning tasks processing completed for this iteration." }
+            }
+        } catch (e: Exception) {
+            logger.error(e) { "Error while processing cleaning task" }
+        }
+    }
+
+    fun processCleaningTask(reservedTask: TaskStepReservationEntryDto): TaskStepResultEntryDto {
+        val businessPartner = reservedTask.businessPartner
```

Comment:
You're only using `businessPartner.generic`. Better store this in `val genericBusinessPartner` ?